### PR TITLE
add tt rocks module

### LINKED ISSUE #14: tt rocks
see `tarantoolctl rocks`

### PR DESCRIPTION
**Pull request by 0x501D**
Add LuaRocks module. LuaRocks embed with Gopher-Lua and embed.FS
features.

Close #14

### CODE REVIEW

File: cli/cmd/rocks.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/tarantool/tt/cli/modules"
+	"github.com/tarantool/tt/cli/rocks"
+)
+
+// NewRocksCmd creates rocks command.
+func NewRocksCmd() *cobra.Command {
+	var rocksCmd = &cobra.Command{
+		Use:                "rocks",
+		Short:              "LuaRocks package manager",
+		DisableFlagParsing: true,
```
Comment (LeonidVas) [inline cli/cmd/rocks.go:16]: Add a few comments on this for future generations.

File: cli/rocks/extra/hardcoded.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
+local function get_cwd()
```
Comment (LeonidVas) [inline cli/rocks/extra/hardcoded.lua:1]: Please add some comments about this file (`hardcoded.lua`).

File: cli/rocks/extra/wrapper.lua [hunk 3]
```diff
@@ -0,0 +1,45 @@
+local function exec(bin, ...)
+    local cfg = require("luarocks.core.cfg")
+    cfg.init()
+    local util = require("luarocks.util")
+    local cmd = require("luarocks.cmd")
+
+    -- Tweak help messages
```
Comment (LeonidVas) [inline cli/rocks/extra/wrapper.lua:7]: Set a dot at the end of the sentence.

File: cli/rocks/extra/wrapper.lua [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    util.see_help = function(command, program) -- luacheck: no unused args
+        return "\nRun \"" .. bin .. " rocks " .. command .. " --help\" for help."
+    end
+    util.this_program = function(default) -- luacheck: no unused args
+        return bin .. " rocks"
+    end
+
+    -- Disabled: path, upload
```
Comment (LeonidVas) [inline cli/rocks/extra/wrapper.lua:15]: Set a dot at the end of the sentence.

File: cli/rocks/rocks.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+	"github.com/tarantool/tt/cli/cmdcontext"
+	lua "github.com/yuin/gopher-lua"
+)
+
+//go:embed extra/*
+//go:embed third_party/luarocks/src/*
+var luarocks embed.FS
+
+func readEmbedFile(f embed.FS, p string) (string, error) {
```
Comment (LeonidVas) [inline cli/rocks/rocks.go:16]: Please do not use one-letter names.

File: cli/rocks/rocks.go [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+	c, err := f.ReadFile(p)
+	if err != nil {
+		return "", err
+	}
+	return string(c), nil
+}
+
+func Exec(cmdCtx *cmdcontext.CmdCtx, args []string) error {
```
Comment (LeonidVas) [inline cli/rocks/rocks.go:24]: Write a description for at least the public functions (preferably all of them).

File: cli/rocks/rocks.go [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+import (
+	"embed"
+	"fmt"
+	"os"
+
+	"github.com/tarantool/tt/cli/cmdcontext"
+	lua "github.com/yuin/gopher-lua"
+)
+
+//go:embed extra/*
```
Comment (LeonidVas) [inline cli/rocks/rocks.go:12]: Set a space after `//`.

File: cli/rocks/rocks.go [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+//go:embed third_party/luarocks/src/*
```
Comment (LeonidVas) [inline cli/rocks/rocks.go:13]: Set a space after `//`.

File: cli/rocks/rocks.go [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+	if err != nil {
+		return "", err
+	}
+	return string(c), nil
+}
+
+func Exec(cmdCtx *cmdcontext.CmdCtx, args []string) error {
+	var cmd string
+
+	/* Print rocks help if no arguments given */
```
Comment (LeonidVas) [inline cli/rocks/rocks.go:27]: Use `//` for comments.
Set a dot at the end of the sentence.

File: cli/rocks/rocks.go [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+	if len(args) == 0 {
+		cmd = "help"
+	}
+
+	for i, s := range args {
```
Comment (LeonidVas) [inline cli/rocks/rocks.go:32]: Please do not use one-letter names.

File: cli/rocks/rocks.go [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+		"luarocks.fetch.hg_ssh":            rocks_path + "luarocks/fetch/hg_ssh.lua",
+		"luarocks.admin.cache":             rocks_path + "luarocks/admin/cache.lua",
+		"luarocks.admin.cmd.refresh_cache": rocks_path + "luarocks/admin/cmd/refresh_cache.lua",
+		"luarocks.admin.index":             rocks_path + "luarocks/admin/index.lua",
+		"luarocks.admin.cmd.add":           rocks_path + "luarocks/admin/cmd/add.lua",
+		"luarocks.admin.cmd.remove":        rocks_path + "luarocks/admin/cmd/remove.lua",
+		"luarocks.admin.cmd.make_manifest": rocks_path + "luarocks/admin/cmd/make_manifest.lua",
+	}
+
+	L := lua.NewState()
```
Comment (LeonidVas) [inline cli/rocks/rocks.go:140]: For `L` one-letter name is OK)

File: README.rst [hunk 12]
```diff
@@ -161,3 +171,4 @@ Common description. For a detailed description, use ``tt help command`` .
 * ``completion`` - generate autocomplete for a specified shell.
 * ``help`` - display help for any command.
 * ``logrotate`` - rotate logs of a started tarantool instance.
+* ``rocks`` - LuaRocks package manager.
```
Comment (LeonidVas) [inline README.rst:174]: This command does not exist, it is created in the next commit. I suggest that you insert the README as the last commit in the patchset.

**Review by LeonidVas — APPROVED** [review]
Hi! Thank you for the patchset.
See several comments bellow.

File: cli/rocks/rocks.go (hunk 7 above)
Comment (0x501D) [reply cli/rocks/rocks.go:12]: This will brake go:embed feature.

File: cli/rocks/rocks.go (hunk 8 above)
Comment (0x501D) [reply cli/rocks/rocks.go:13]: This will brake go:embed feature.

File: README.rst (hunk 12 above)
Comment (0x501D) [reply README.rst:174]: fixed

File: cli/rocks/rocks.go (hunk 9 above)
Comment (0x501D) [reply cli/rocks/rocks.go]: fixed

File: cli/rocks/rocks.go (hunk 6 above)
Comment (0x501D) [reply cli/rocks/rocks.go:25]: Description added

File: cli/rocks/rocks.go (hunk 5 above)
Comment (0x501D) [reply cli/rocks/rocks.go]: Fixed

File: cli/rocks/extra/wrapper.lua (hunk 4 above)
Comment (0x501D) [reply cli/rocks/extra/wrapper.lua]: Fixed

File: cli/rocks/extra/wrapper.lua (hunk 3 above)
Comment (0x501D) [reply cli/rocks/extra/wrapper.lua]: Fixed

File: cli/rocks/extra/hardcoded.lua (hunk 2 above)
Comment (0x501D) [reply cli/rocks/extra/hardcoded.lua:3]: Added

**Pull request merged**